generators/pkg/generator/generate_test.go

```diff
@@ -6,33 +6,285 @@ import (
 	"github.com/google/go-cmp/cmp"
 	v1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
 	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 func TestGenerateTask(t *testing.T) {
 	want := v1beta1.Task{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: v1beta1.SchemeGroupVersion.String(),
+			Kind:       "Task",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "github-build",
+			Labels: map[string]string{"generator.tekton.dev": "github-build"},
+		},
 		Spec: v1beta1.TaskSpec{
-			Workspaces: []v1beta1.WorkspaceDeclaration{{Name: "input",
-				MountPath: "/input"}},
+			Workspaces: []v1beta1.WorkspaceDeclaration{
+				{
+					Name:      "input",
+					MountPath: "/input",
+				},
+			},
+			Steps: []v1beta1.Step{
+				{
+					Container: corev1.Container{
+						Name:    "build",
+						Image:   "gcr.io/kaniko-project/executor:latest",
+						Command: []string{"/kaniko/executor"},
+						Args: []string{"--context=dir://$(workspaces.input.path)/src",
+							"--destination=gcr.io/wlynch-test/kaniko-test",
+							"--verbosity=debug"},
+					},
+				},
+			},
+		},
+	}
+	github := GitHub{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "github-build",
+		},
+		Spec: GitHubSpec{
+			URL: "https://github.com/wlynch/test",
 			Steps: []v1beta1.Step{{Container: corev1.Container{
 				Name:    "build",
 				Image:   "gcr.io/kaniko-project/executor:latest",
 				Command: []string{"/kaniko/executor"},
 				Args: []string{"--context=dir://$(workspaces.input.path)/src",
 					"--destination=gcr.io/wlynch-test/kaniko-test",
 					"--verbosity=debug"},
-			}}}}}
-	spec := GitHubSpec{
-		URL: "https://github.com/wlynch/test",
-		Steps: []v1beta1.Step{{Container: corev1.Container{
-			Name:    "build",
-			Image:   "gcr.io/kaniko-project/executor:latest",
-			Command: []string{"/kaniko/executor"},
-			Args: []string{"--context=dir://$(workspaces.input.path)/src",
-				"--destination=gcr.io/wlynch-test/kaniko-test",
-				"--verbosity=debug"},
-		}}}}
-	got := GenerateTask(spec)
+			},
+			}}}}
+	got := GenerateTask(github)
 	if diff := cmp.Diff(want, got); diff != "" {
 		t.Errorf("Tasks mismatch (-want +got):\n %s", diff)
 	}
 }
+
+func TestAddGitCloneTask(t *testing.T) {
```

Comment:
I don't think this is a useful test. This is prone to breaking if there is any upstream change to either of the tasks. I mentioned in another comment to not fetch remote configs and just rely on what's on the cluster. Once you do that this can be removed.